install/kubernetes/cilium/values.yaml

```diff
@@ -497,8 +497,14 @@ global:
     # Set this field ":4244" if you are enabling hubble-relay, as it assumes that Hubble is listening
     # on port 4244.
     listenAddress: ""
+    tls:
+      # When set to true, automatically generate a CA and certificates to
+      # enable mTLS between Hubble server and Hubble Relay.
+      # If set to false, the certs for Hubble server and Hubble Relay need to
+      # be provided by setting appropriate values in the hubble-tls chart.
+      auto: true
```

Comment:
I wonder if we should use `auto: 'helm'` here instead of a bool. We might end up with other ways to auto-generate certificates.

Or to ask differently: If we end up having a K8s Job or Deployment to generate Hubble certs as an alternative to Helm, how would the Helm values look in that case.

Reply:
That's a good point that you raise. I feel that it would be weird to have something else than a boolean associated to "auto" though.

What about:
```
auto:
  enabled: true
  method: 'helm'
```

If we keep `auto: bool`, we may also add a new key/value pairs to specify the method at a later point, should the need arise.

Reply:
TBH I thought the `auto` mode was automatically being done by our code, not helm. So yeah, it makes sense to specify the method as 'helm'

Reply:
I gave this more thoughts: I think my proposition above is better as this could allow specifying things such as the generated cert validity duration in days or other options such as the type of keys, etc.

Reply:
I added a commit to address this and also make the cert validity duration configurable:
```yaml
    tls:
      # Enable mutual TLS for listenAddress. Setting this value to false is
      # highly discouraged.
      enabled: false
      # Configure automatic TLS certificates generation.
      auto:
        # When set to true, automatically generate a CA and certificates to
        # enable mTLS between Hubble server and Hubble Relay. If set to false,
        # the certs for Hubble server need to be provided by setting
        # appropriate values in the hubble-tls chart.
        enabled: true
        # Sets the method to auto-generate certificates. Currently, only "helm"
        # is supported. This method uses Helm to generate certificates.
        method: "helm"
        # Generated certificates validity duration in days.
        certValidityDuration: 1095
```